src/libraries/System.Net.Requests/src/System/Net/RequestStream.cs

```diff
@@ -45,17 +55,34 @@ public override bool CanWrite
             }
         }
 
+        // We're already sending data as StreamContent,
+        // so we buffer data in memory and flush it in three cases:
+        // - When GetResponse called.
+        // - When RequestStream is getting disposed.
+        // - When user calls Flush.
         public override void Flush()
         {
-            // Nothing to do.
+            ThrowIfDisposed();
+
+            if (_isBuffered && _buffer!.Length > 0)
```

Comment:
If we always write into the stream buffer, the flushing logic also becomes unnecessary